src/services/review/ReviewRequestService.ts

```diff
@@ -0,0 +1,365 @@
+import { ModelStatic } from "sequelize"
+
+import UserWithSiteSessionData from "@classes/UserWithSiteSessionData"
+
+import { Reviewer } from "@database/models/Reviewers"
+import { ReviewMeta } from "@database/models/ReviewMeta"
+import { ReviewRequest } from "@database/models/ReviewRequest"
+import { Site } from "@root/database/models/Site"
+import { User } from "@root/database/models/User"
+import RequestNotFoundError from "@root/errors/RequestNotFoundError"
+import {
+  DashboardReviewRequestDto,
+  EditedItemDto,
+  FileType,
+  ReviewRequestDto,
+} from "@root/types/dto /review"
+import { Commit, fromGithubCommitMessage } from "@root/types/github"
+import { RequestChangeInfo } from "@root/types/review"
+import * as ReviewApi from "@services/db/review"
+
+/**
+ * NOTE: This class does not belong as a subset of GitHub service.
+ * This is because GitHub service exists to operate on _files_
+ * whereas this operates on pull requests.
+ *
+ * Perhaps we could rename Github service into GitHubFile service
+ * and this into GitHubPullRequest service to make the distinction obvious.
+ *
+ * Separately, this also allows us to add typings into this service.
+ */
+export default class ReviewRequestService {
+  private readonly apiService: typeof ReviewApi
+
+  private readonly repository: ModelStatic<ReviewRequest>
+
+  private readonly users: ModelStatic<User>
+
+  private readonly reviewers: ModelStatic<Reviewer>
+
+  private readonly reviewMeta: ModelStatic<ReviewMeta>
+
+  constructor(
+    apiService: typeof ReviewApi,
+    users: ModelStatic<User>,
+    repository: ModelStatic<ReviewRequest>,
+    reviewers: ModelStatic<Reviewer>,
+    reviewMeta: ModelStatic<ReviewMeta>
+  ) {
+    this.apiService = apiService
+    this.users = users
+    this.repository = repository
+    this.reviewers = reviewers
+    this.reviewMeta = reviewMeta
+  }
+
+  compareDiff = async (
+    sessionData: UserWithSiteSessionData
+  ): Promise<EditedItemDto[]> => {
+    // Step 1: Get the site name
+    const { siteName } = sessionData
+
+    // Step 2: Get the list of changed files using Github's API
+    // Refer here for details; https://docs.github.com/en/rest/commits/commits#compare-two-commits
+    // Note that we need a triple dot (...) between base and head refs
+    const { files, commits } = await this.apiService.getCommitDiff(siteName)
+
+    const mappings = await this.computeShaMappings(commits)
+
+    return files.map(({ filename, contents_url }) => {
+      const fullPath = filename.split("/")
+      const items = contents_url.split("?ref=")
+      // NOTE: We have to compute sha this way rather than
```

Comment:
Minor nit: I think this comment is incomplete? Might need more explanation on what this sha is referring to (I got confused here when reviewing hahahaha).

Reply:
Oh oof I think the comment got deleted through rebase